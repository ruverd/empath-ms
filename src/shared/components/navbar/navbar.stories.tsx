import type { Meta, StoryObj } from "@storybook/react";
import { BrowserRouter } from "react-router-dom";

import { Navbar } from "./navbar";

const meta = {
  title: "Components/Navbar",
  component: Navbar,
  parameters: {
    layout: "fullscreen",
  },
  tags: ["autodocs"],
  decorators: [
    (Story) => (
      <BrowserRouter>
        <Story />
      </BrowserRouter>
    ),
  ],
} satisfies Meta<typeof Navbar>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  render: () => <Navbar />,
};
