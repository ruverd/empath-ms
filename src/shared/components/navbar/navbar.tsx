import { Menu } from "lucide-react";
import { useState } from "react";
import { Link } from "react-router-dom";

import { Logo } from "../logo";
import { Button } from "../ui/button";
import { Sheet, SheetContent } from "../ui/sheet";
import { Nav } from "./components/nav";
import { UserActions } from "./components/user-actions";

export const Navbar = () => {
  const [isOpen, setIsOpen] = useState(false);

  const handleClose = () => {
    setIsOpen(false);
  };

  return (
    <div className="bg-background shadow-dark px-6 h-[68px] flex items-center">
      <div className="mx-auto flex justify-between items-center w-full">
        <div className="flex items-center gap-[50px]">
          <Link to="/">
            <Logo />
          </Link>
          <Nav className="hidden lg:block" />
        </div>
        <div className="hidden lg:block">
          <UserActions />
        </div>
        <Button
          variant="ghost"
          size="icon"
          className="lg:hidden"
          onClick={() => setIsOpen(true)}
        >
          <Menu className="size-6" />
          <span className="sr-only">Toggle Menu</span>
        </Button>
      </div>

      <Sheet open={isOpen} onOpenChange={setIsOpen}>
        <SheetContent
          side="right"
          className="w-[300px] sm:w-[400px] p-0 rounded-tl-2xl rounded-bl-2xl"
        >
          <div className="flex flex-col h-full">
            <div className="p-6">
              <Nav onNavigate={handleClose} />
            </div>
            <div className="mt-auto border-t p-6">
              <UserActions onAction={handleClose} />
            </div>
          </div>
        </SheetContent>
      </Sheet>
    </div>
  );
};
