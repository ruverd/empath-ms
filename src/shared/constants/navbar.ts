export const NAV_ITEMS = [
  {
    label: "Dashboard",
    href: "/",
  },
  {
    label: "Members",
    href: "/members",
  },
  {
    label: "Courses",
    href: "/courses",
  },
  {
    label: "Learning Pathways",
    href: "/learning",
  },
  {
    label: "Live Streams",
    href: "/live",
  },
];
