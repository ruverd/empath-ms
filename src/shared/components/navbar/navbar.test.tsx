import { fireEvent, render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { describe, expect, it, vi } from "vitest";

import { Navbar } from "./navbar";

vi.mock("./components/nav", () => ({
  Nav: ({ onNavigate }: { onNavigate?: () => void }) => (
    <div data-testid="nav" onClick={onNavigate}>
      Nav Component
    </div>
  ),
}));

vi.mock("./components/user-actions", () => ({
  UserActions: ({ onAction }: { onAction?: () => void }) => (
    <div data-testid="user-actions" onClick={onAction}>
      User Actions Component
    </div>
  ),
}));

vi.mock("../logo", () => ({
  Logo: () => <div data-testid="logo">Logo Component</div>,
}));

const renderNavbar = () => {
  return render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>
  );
};

describe("Navbar", () => {
  it("should match snapshot", () => {
    const { container } = renderNavbar();
    expect(container).toMatchSnapshot();
  });

  it("should render all components correctly", () => {
    renderNavbar();

    expect(screen.getByTestId("logo")).toBeInTheDocument();
    expect(screen.getByTestId("nav")).toBeInTheDocument();
    expect(screen.getByTestId("user-actions")).toBeInTheDocument();
    expect(
      screen.getByRole("button", { name: /toggle menu/i })
    ).toBeInTheDocument();
  });

  it("should open mobile menu when toggle button is clicked", () => {
    renderNavbar();

    const toggleButton = screen.getByRole("button", { name: /toggle menu/i });
    fireEvent.click(toggleButton);

    expect(screen.getAllByTestId("nav")).toHaveLength(2);
    expect(screen.getAllByTestId("user-actions")).toHaveLength(2);
  });

  it("should close mobile menu when navigation occurs", () => {
    renderNavbar();

    const toggleButton = screen.getByRole("button", { name: /toggle menu/i });
    fireEvent.click(toggleButton);

    const mobileNav = screen.getAllByTestId("nav")[1];
    fireEvent.click(mobileNav);

    expect(screen.getAllByTestId("nav")).toHaveLength(1);
    expect(screen.getAllByTestId("user-actions")).toHaveLength(1);
  });
});
