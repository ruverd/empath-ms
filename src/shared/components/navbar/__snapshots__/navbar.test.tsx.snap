// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar > should match snapshot 1`] = `
<div>
  <div
    class="bg-background shadow-dark px-6 h-[68px] flex items-center"
  >
    <div
      class="mx-auto flex justify-between items-center w-full"
    >
      <div
        class="flex items-center gap-[50px]"
      >
        <a
          href="/"
        >
          <div
            data-testid="logo"
          >
            Logo Component
          </div>
        </a>
        <div
          data-testid="nav"
        >
          Nav Component
        </div>
      </div>
      <div
        class="hidden lg:block"
      >
        <div
          data-testid="user-actions"
        >
          User Actions Component
        </div>
      </div>
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-[40px] text-sm font-bold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-transparent hover:bg-accent hover:text-primary-cloud h-10 w-10 lg:hidden"
      >
        <svg
          class="lucide lucide-menu size-6"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="4"
            x2="20"
            y1="12"
            y2="12"
          />
          <line
            x1="4"
            x2="20"
            y1="6"
            y2="6"
          />
          <line
            x1="4"
            x2="20"
            y1="18"
            y2="18"
          />
        </svg>
        <span
          class="sr-only"
        >
          Toggle Menu
        </span>
      </button>
    </div>
  </div>
</div>
`;
